iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,339 @@
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    // i=1 because there is already an active runtime in test fixture class which acquired an application port
+    for (auto i = 1u; i < iox::MAX_PROCESS_NUMBER; ++i)
+    {
+        auto appPort = m_runtime->getMiddlewareApplication();
+        EXPECT_NE(nullptr, appPort);
+    }
+
+    auto appPort = m_runtime->getMiddlewareApplication();
+
+    EXPECT_EQ(nullptr, appPort);
+    EXPECT_TRUE(errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareInterfaceIsSuccessful)
+{
+    const auto interfacePortData = m_runtime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL, m_runnableName);
+
+    EXPECT_EQ(m_runtimeName, interfacePortData->m_processName);
+    EXPECT_EQ(iox::capro::ServiceDescription(0u, 0u, 0u), interfacePortData->m_serviceDescription);
+    EXPECT_EQ(false, interfacePortData->m_toBeDestroyed);
+    EXPECT_EQ(true, interfacePortData->m_doInitialOfferForward);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareInterface_InterfacelistOverflow)
+{
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    for (auto i = 0u; i < iox::MAX_INTERFACE_NUMBER + 1u; ++i)
+    {
+        m_runtime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL);
+    }
+
+    EXPECT_TRUE(errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_ValidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << m_runtimeName
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_runnableName;
+
+    const auto successfullySent = m_runtime->sendMessageToRouDi(m_sendBuffer);
+
+    EXPECT_EQ(true, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_InvalidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << std::string()
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_invalidRunnableName;
+
+    const auto successfullySent = m_runtime->sendMessageToRouDi(m_sendBuffer);
+
+    EXPECT_EQ(false, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_EmptyMessage)
+{
+    const auto successfullySent = m_runtime->sendMessageToRouDi(m_sendBuffer);
+
+    EXPECT_EQ(true, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, SendRequestToRouDi_ValidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << m_runtimeName
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_runnableName;
+
+    const auto successfullySent = m_runtime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
```

Comment:
the `getMiddlewareX` methods are using this method to get the ports, so basically if those tests work, this will also work. I think the only reason this method is public is because we have users who might extend RouDi and send custom messages